sdk/Lusid.Sdk/Utilities/LusidExceptionHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using Lusid.Sdk.Client;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Handles the generation of LUSID exceptions after receiving the ApiResponse
+    /// </summary>
+    public static class LusidExceptionHandler
+    {
+        /// <summary>
+        /// Generate exceptions from the ApiResponse when ResponseStatus is an Error,
+        /// and StatusCode has no content or is less than 400
+        /// </summary>
+        /// <param name="methodName">The name of the method</param>
+        /// <param name="response">The ApiResponse</param>
+        public static Exception CustomExceptionFactory(string methodName, IApiResponse response)
+        {
+            
+            var status = (int) response.StatusCode; // Http status code
+            var errorText = response.ErrorText; // Error text attached on exceptions
+            var headers = response.Headers; // Http headers that come back with the api response
+            var responseBodyString = response.RawContent; // The body content of the http response as a string
+            
+            // Use default exception, and only use subsequent checks if default exception factory is null.
+            var defaultException = Configuration.DefaultExceptionFactory.Invoke(methodName, response);
+            if (defaultException != null) return defaultException;
+            
+            // Throw whenever an internal SDK error has been thrown.
+            // Internal SDK deserialization errors will result in ErrorText to be not null.
+            if (response.ErrorText != null)
```

Comment:
Are we using != null or IsNullOrWhiteSpace? Let's pick one to use for both

Reply:
what's the problem with != null?

Reply:
Perhaps it is helpful to also check if it is just white space, but in this case it sounds like it should always be null if not an error so let's leave it as is.